nomad/structs/structs.go

```diff
@@ -2090,6 +2090,14 @@ type Node struct {
 	// LastDrain contains metadata about the most recent drain operation
 	LastDrain *DrainMetadata
 
+	// LastMissedHeartbeatIndex stores the Raft index when the node
+	// last missed a heartbeat.
+	LastMissedHeartbeatIndex uint64
```

Comment:
I'm not sure how accurate this name is. It's more like the first time the node transition to an unresponsive status, but I couldn't think of a good name for this 😅

Reply:
Ah good point. Resetting it zero may help to indicate that this field is sort edge triggered. I pushed a commit do just that. Thanks!